# Unit test for find_outliers

In [1]:
import pandas as pd

from test_data_utilities_find_outlier_qrcodes import prepare_test_df, test_find_outlier_qrcodes_age_min, test_find_outlier_qrcodes_age_max, test_find_outlier_qrcodes_weight_min, test_find_outlier_qrcodes_weight_max, test_find_outlier_qrcodes_height_min, test_find_outlier_qrcodes_height_max

In [2]:
df = prepare_test_df()

In [3]:
df.head()

artifacts   age  weight  height  \
0  scans/1585013006-yqwb95138e/100/pc_1585013006-...   180     3.0    30.0   
1  scans/1555555555-yqqqqqqqqq/100/pc_1555555555-...   365     6.0    95.0   
2  scans/1212121212-jajajajaja/100/pc_1212121212-...  3000    32.0   102.0   
3  scans/1585013006-yqwb95138e/100/pc_1585013006-...   180     3.0    30.0   
4  scans/9922992299-lelelelele/100/pc_9922992299-...    20     2.7   195.0   

                  qrcode  
0  1585013006-yqwb95138e  
1  1555555555-yqqqqqqqqq  
2  1212121212-jajajajaja  
3  1585013006-yqwb95138e  
4  9922992299-lelelelele

### Test age min <182 - should be 2 qrcodes [1585013006-yqwb95138e (duplicate) and 9922992299-lelelelele]

In [4]:
test_find_outlier_qrcodes_age_min(df)

2021-04-13 09:48:11,388 - INFO - Running the following query: @df.age<365/2 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 59
2021-04-13 09:48:11,397 - INFO - NumExpr defaulting to 4 threads. - /anaconda/envs/azureml_py36/lib/python3.6/site-packages/numexpr/utils.py: line 157
2021-04-13 09:48:11,406 - INFO - No. of unusual artifacts: 3 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 61
2021-04-13 09:48:11,408 - INFO - No. of unique outlier qr_codes: 2 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 63
2021-04-13 09:48:11,409 - INFO - Extracting qr_codes... - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 64
2

No. of qrcodes: %d 2


['1585013006-yqwb95138e', '9922992299-lelelelele']

### Test age max <2190 - should be 1 qrcode [1212121212-jajajajaja]

In [5]:
test_find_outlier_qrcodes_age_max(df)

2021-04-13 09:48:11,418 - INFO - Running the following query: @df.age>365*6 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 59
2021-04-13 09:48:11,422 - INFO - No. of unusual artifacts: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 61
2021-04-13 09:48:11,426 - INFO - No. of unique outlier qr_codes: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 63
2021-04-13 09:48:11,426 - INFO - Extracting qr_codes... - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 64
2021-04-13 09:48:11,427 - INFO - No. of qrcodes: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data

No. of qrcodes: %d 1


['1212121212-jajajajaja']

### Test weight min <5.0 - should be 2 qrcodes [1585013006-yqwb95138e (duplicate) and 9922992299-lelelelele]

In [6]:
test_find_outlier_qrcodes_weight_min(df)

2021-04-13 09:48:11,451 - INFO - Running the following query: @df.weight<5.0 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 59
2021-04-13 09:48:11,457 - INFO - No. of unusual artifacts: 3 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 61
2021-04-13 09:48:11,460 - INFO - No. of unique outlier qr_codes: 2 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 63
2021-04-13 09:48:11,461 - INFO - Extracting qr_codes... - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 64
2021-04-13 09:48:11,462 - INFO - No. of qrcodes: 2 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/dat

No. of qrcodes: %d 2


['1585013006-yqwb95138e', '9922992299-lelelelele']

### Test weight max >30.0 - should be 1 qrcode [1212121212-jajajajaja]

In [7]:
test_find_outlier_qrcodes_weight_max(df)

2021-04-13 09:48:11,468 - INFO - Running the following query: @df.weight>30.0 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 59
2021-04-13 09:48:11,474 - INFO - No. of unusual artifacts: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 61
2021-04-13 09:48:11,476 - INFO - No. of unique outlier qr_codes: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 63
2021-04-13 09:48:11,476 - INFO - Extracting qr_codes... - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 64
2021-04-13 09:48:11,478 - INFO - No. of qrcodes: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/da

No. of qrcodes: %d 1


['1212121212-jajajajaja']

### Test height min <40.0 cm - should be 1 qrcode [1585013006-yqwb95138e (duplicate)]

In [8]:
test_find_outlier_qrcodes_height_min(df)

2021-04-13 09:48:11,485 - INFO - Running the following query: @df.height<40.0 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 59
2021-04-13 09:48:11,488 - INFO - No. of unusual artifacts: 2 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 61
2021-04-13 09:48:11,491 - INFO - No. of unique outlier qr_codes: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 63
2021-04-13 09:48:11,491 - INFO - Extracting qr_codes... - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 64
2021-04-13 09:48:11,492 - INFO - No. of qrcodes: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/da

No. of qrcodes: %d 1


['1585013006-yqwb95138e']

### Test height max >150.0 cm - should be 1 qrcode [9922992299-lelelelele]

In [9]:
test_find_outlier_qrcodes_height_max(df)

2021-04-13 09:48:11,499 - INFO - Running the following query: @df.height>150.0 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 59
2021-04-13 09:48:11,502 - INFO - No. of unusual artifacts: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 61
2021-04-13 09:48:11,505 - INFO - No. of unique outlier qr_codes: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 63
2021-04-13 09:48:11,505 - INFO - Extracting qr_codes... - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/data_utils/dataset_EDA/data_utilities.py: line 64
2021-04-13 09:48:11,506 - INFO - No. of qrcodes: 1 - /mnt/resource/batch/tasks/shared/LS_root/mounts/clusters/jzcomp/code/Users/jziegler/cgm-ml/src/d

No. of qrcodes: %d 1


['9922992299-lelelelele']